src/ToDo.js

```diff
@@ -0,0 +1,68 @@
+// IDEA Impose length limit, add deletable tasks, store data in browser for preserverence.
+// DONE Completed assignment and added toggleable checkbox/cross-through, added duplicate checker.
+
+import React, { Component } from 'react';
+import List from './List';
+
+const previousToDos = []; // Storing previous ToDo's in an array so I can easily search for duplicates.
+
+class ToDoList extends Component { // Create a new Component called ToDoList.
+  constructor() {
+    super();
+    this.state = {
+      newToDo: '',
+      ToDos: []
+    };
+
+    this.updateNewToDo = this.updateNewToDo.bind(this); // TODO not sure what this does exactly.
+    this.addToDo = this.addToDo.bind(this);
+  }
+
+  updateNewToDo(event) {
+    this.setState({
+      newToDo: event.target.value
+    });
+  }
+
+  addToDo(event) {
+    event.preventDefault(); // Disable the default behavior of refreshing the page upon clicking enter.
+    document.getElementById("mainInput").value = ""; // Clear input field.
+
+    if (!previousToDos.includes(this.state.newToDo)) { // Check for duplicates.
+      console.log(`Adding '${this.state.newToDo}' to the list.`);
+      previousToDos.push(this.state.newToDo);
+
+      if (document.getElementById('response').innerHTML) document.getElementById('response').innerHTML = ''; // If the response box needs to be cleared, do so.
```

Comment:
once again there are easier patterns for selecting elements in React vs accessing the DOM directly.